connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -384,6 +452,40 @@ public void onCompletion(RecordMetadata recordMetadata, Exception e) {
         return true;
     }
 
+    // Due to transformations that may change the destination topic of a record (such as
+    // RegexRouter) topic creation can not be batched for multiple topics
+    private void maybeCreateTopic(String topic) {
+        if (!topicCreation.isTopicCreationEnabled() || topicCreation.topicCache().contains(topic)) {
+            return;
+        }
```

Comment:
Again, this logic could be moved into `TopicCreation` for easier testing and to make this code more readable:
```suggestion
        if (!topicCreation.isCreateTopicRequired(topic)) {
            return;
        }
```